cmd/usage-stats-handler/main.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"flag"
+	"io/ioutil"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+
+	"github.com/Jeffail/gabs"
+	"github.com/cortexproject/cortex/pkg/util"
+	"github.com/cortexproject/cortex/pkg/util/flagext"
+	"github.com/go-kit/kit/log/level"
+	"github.com/grafana/usage-stats-handler/pkg/metricsutil"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/version"
+	"github.com/weaveworks/common/server"
+)
+
+func main() {
+	var (
+		serverConfig          server.Config
+		instrumentationConfig metricsutil.InstrumentationServerConfig
+		downstreamUrlFlag     string
+		applicationName       = "usage_stats_handler"
+	)
+
+	flagext.RegisterFlags(&serverConfig, &instrumentationConfig)
+	flag.StringVar(&downstreamUrlFlag, "usage-stats.downstream", "http://localhost:3000", "URL of the original usage stats handler")
+	flag.Parse()
+
+	util.InitLogger(&serverConfig)
+
+	downstreamUrl, err := url.Parse(downstreamUrlFlag)
+	if err != nil {
+		level.Error(util.Logger).Log("msg", "failed to create server", "err", err)
+		return
+	}
+
+	server, err := server.New(serverConfig)
+	if err != nil {
+		level.Error(util.Logger).Log("msg", "failed to create server", "err", err)
+		return
+	}
+	defer server.Shutdown()
+
+	server.HTTP.PathPrefix("/").Name("proxydownstream").HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		logPayload(r)
+		httputil.NewSingleHostReverseProxy(downstreamUrl).ServeHTTP(w, r)
+	})
+
+	prometheus.MustRegister(version.NewCollector(applicationName))
+
+	instrumentationServer, err := metricsutil.NewInstrumentationServer(instrumentationConfig)
+	if err != nil {
+		level.Error(util.Logger).Log("msg", "error initializing the instrumentation server", "err", err)
+		return
+	}
+	defer instrumentationServer.Stop()
+
+	server.Run()
+}
+
+func logPayload(req *http.Request) {
+	copy := req.Clone(req.Context())
+
+	body, err := ioutil.ReadAll(copy.Body)
+	if err != nil {
+		level.Error(util.Logger).Log("failed to read body", "error", err)
+	}
+
+	jsonParsed, err := gabs.ParseJSON(body)
+	if err != nil {
+		level.Error(util.Logger).Log("failed to parse json", "error", err)
+	}
+
+	level.Info(util.Logger).Log("usage payload", jsonParsed)
```

Comment:
Not sure about the convention here, but it seems like this should be more like:
```suggestion
		level.Error(util.Logger).Log("msg", "failed to read body", "error", err)
	}

	jsonParsed, err := gabs.ParseJSON(body)
	if err != nil {
		level.Error(util.Logger).Log("msg", "failed to parse json", "error", err)
	}

	level.Info(util.Logger).Log("msg", "usage payload", "payload", jsonParsed)
```

Reply:
the payload output looks like this 
```
level=info ts=2021-05-05T07:59:17.968736237Z caller=main.go:77 usagepayload="{\"arch\":\"amd64\",\"edition\":\"oss\",\"hasValidLicense\":false,\"metrics\":{\"stats.active_users.count\":0,\"stats.alerting.ds.other.count\":0,\"stats.alerts.count\":0,\"stats.annotations.count\":0,\"stats.auth_enabled.anonymous.count\":0,\"stats.auth_enabled.auth_proxy.count\":0,\"stats.auth_enabled.basic_auth.count\":1,\"stats.auth_enabled.ldap.count\":0,\"stats.auth_enabled.oauth_azuread.count\":0,\"stats.auth_enabled.oauth_generic_oauth.count\":0,\"stats.auth_enabled.oauth_github.count\":0,\"stats.auth_enabled.oauth_gitlab.count\":0,\"stats.auth_enabled.oauth_google.count\":0,\"stats.auth_enabled.oauth_grafana_com.count\":0,\"stats.auth_enabled.oauth_okta.count\":0,\"stats.auth_token_per_user_le_12\":0,\"stats.auth_token_per_user_le_15\":0,\"stats.auth_token_per_user_le_3\":0,\"stats.auth_token_per_user_le_6\":0,\"stats.auth_token_per_user_le_9\":0,\"stats.auth_token_per_user_le_inf\":0,\"stats.avg_auth_token_per_user.count\":0,\"stats.dashboard_permissions.count\":0,\"stats.dashboard_versions.count\":64,\"stats.dashboards.count\":61,\"stats.datasources.count\":0,\"stats.distributor.grafana-labs.count\":1,\"stats.ds.other.count\":0,\"stats.edition.enterprise.count\":0,\"stats.edition.oss.count\":1,\"stats.folder_permissions.count\":0,\"stats.folders.count\":1,\"stats.orgs.count\":1,\"stats.packaging.dev.count\":1,\"stats.playlist.count\":0,\"stats.plugins.apps.count\":0,\"stats.plugins.datasources.count\":20,\"stats.plugins.panels.count\":24,\"stats.provisioned_dashboards.count\":61,\"stats.snapshots.count\":0,\"stats.stars.count\":0,\"stats.teams.count\":0,\"stats.total_auth_token.count\":0,\"stats.users.count\":1,\"stats.valid_license.count\":0},\"os\":\"linux\",\"packaging\":\"dev\",\"version\":\"7_5_0-pre\"}"
```

so I don't think we need `msg` but I think we need to convert the payload to logfmt/json only to make it easy for Loki to consume.